packages/ocamlbuild/ocamlbuild.0.14.2+win/opam

```diff
@@ -0,0 +1,38 @@
+opam-version: "2.0"
+synopsis:
+  "OCamlbuild is a build system with builtin rules to easily build most OCaml projects"
+maintainer: "Gabriel Scherer <gabriel.scherer@gmail.com>"
+authors: ["Nicolas Pouillard" "Berke Durak"]
+license: "LGPL-2.0-or-later WITH OCaml-LGPL-linking-exception"
+homepage: "https://github.com/ocaml/ocamlbuild/"
+doc: "https://github.com/ocaml/ocamlbuild/blob/master/manual/manual.adoc"
+bug-reports: "https://github.com/ocaml/ocamlbuild/issues"
+depends: [
+  "ocaml" {>= "4.03"}
+]
+conflicts: [
+  "base-ocamlbuild"
+  "ocamlfind" {< "1.6.2"}
+]
+build: [
+  [make "all"]
+]
+install: [
+  [make "install"]
+  ["mkdir" "-p" "%{lib}%/ocamlbuild"]
+  ["install" "-m" "0644" "META" "%{lib}%/ocamlbuild"]
+]
+dev-repo: "git+https://github.com/ocaml/ocamlbuild.git"
+url {
+  src: "https://github.com/ocaml/ocamlbuild/archive/refs/tags/0.14.2.tar.gz"
+  checksum: [
+    "md5=2f407fadd57b073155a6aead887d9676"
+    "sha512=f568bf10431a1f701e8bd7554dc662400a0d978411038bbad93d44dceab02874490a8a5886a9b44e017347e7949997f13f5c3752f74e1eb5e273d2beb19a75fd"
+  ]
+}
+extra-source "ocamlbuild-0.14.2.patch" {
+  src: "https://raw.githubusercontent.com/ocaml-opam/opam-repository-mingw/354a87b397856f2a70024c5c83fc5001074935b6/packages/ocamlbuild/ocamlbuild.0.14.2/files/ocamlbuild-0.14.2.patch"
```

Comment:
should we have this file directly in the tree instead?

Reply:
exactly the [opposite review above](https://github.com/ocaml/opam-repository/pull/23832/files#diff-6728a620d9aaae70089e9099543035a7522657d5b5753414e0bc4590b0de41b6) from @dra27 ;-) By holding it externally, we dont hold a fairly large patch in opam-repo, so I think external is better in this case.

Reply:
The only downside of external sources is if they disappear (this happened with a very old compiler patches, IIRC), but in general in-tree is not great for patches both because it forces _all_ users to download them and because often the patches end up being applied to more than one package, so they get duplicated on users' machines as well.

Reply:
Or if GitHub/git changes the patch files (cf. #23837) but note that this change here isn't affected because we're referencing a patch file in a repo, rather than a commit.

Reply:
What if we move it to the opam-source-archive so that at least we control the repo where it comes from?

Reply:
Indeed, it's in ocaml-opam/opam-repository-mingw rather than fdopen's. We could move it - I did have in mind many months ago to put a branch for patches in opam-repository itself, but I haven't moved forward with that proposal. ocaml-opam/opam-repository-mingw will hopefully at some point be "archived", but it should never be deleted.